src/game_controller.py

```diff
@@ -36,15 +36,7 @@ def __init__(self):
         self.bot = None
         self.char_selector = None
 
-    def run_bot(self, pick_corpse: bool = False):
-        if self._config.general['restart_d2r_when_stuck']:
```

Comment:
After moving this to Bot(), if the d2 restarts, the saved char template does not persist.

Reply:
Do you think this issue can be solved? This breaks the save game recovery if for any reason the character you want isn't the top-most online character. I have already implemented improvements to the character selector, but I don't know how to make the result of CharSelector.save_char_template() persist after a new instance of Bot() is created. Is there a way to save the template in game_controller.py's scope?

Reply:
I see what you mean now. I'll check later.

Reply:
I've move it to a class variable. I've modified the __main__ code in char_selector.py. It's able to select the character after I moved the window in between the instances of CharSelector. It's not the greatest way to solve this problem but at least it's encapsulated enough for us to easily change this in the future